catt/stream_info.py

```diff
@@ -53,9 +53,9 @@ def __init__(self, video_url, host=None, model=None, device_type=None, ytdl_opti
                 # We pop the "format" item, as it will make get_stream_info fail,
                 # if it holds an invalid value.
                 self._best_format = self._ydl.params.pop("format")
-            elif device_type in AUDIO_DEVICE_TYPES:
+            elif device_type and device_type in AUDIO_DEVICE_TYPES:
                 self._best_format = AUDIO_FORMAT
-            elif model in ULTRA_MODELS:
+            elif model and model in ULTRA_MODELS:
```

Comment:
What was the problem here that you had to add the `model and` part? I can't see any way for this to fail.

Reply:
You're right, I just like only going forward with the check if ```model``` actually holds a value (```None``` is a valid key in a python dict).

Reply:
Ah, right, that's why I was confused (because `None` isn't in `ULTRA_MODELS` so the check is not strictly necessary). Looks good!